item.rb

```diff
@@ -3,10 +3,13 @@
 class Item
   attr_accessor :publish_date
 
-  def initialize(id, publish_date, archived: false)
+  def initialize(id = Random.rand(1..1000))
     @id = id
-    @publish_date = Date.parse(publish_date) # publish_date should be a string = 'DD-MM-YYYY' or 'DD/MM/YYYY'
-    @archived = archived
+    @genre = nil
+    @author = nil
+    @label = nil
+    @publish_date = nil
+    @archived = false
   end
```

Comment:
![image](https://user-images.githubusercontent.com/42799579/188540032-4d68f80a-1f78-49e9-9358-0060deefca5f.png)
Please check the exception, these properties shouldn't be on the constructor. Kindly use the setter methods.

Reply:
Updated. 
And created custom setter methods for them. [link ](https://github.com/Gopxfs/catalog-of-my-things/pull/51/commits/97e356aa0f76d90a4a56b593761f6db765a07536#r963509766)